OpenRA.Game/FieldSaver.cs

```diff
@@ -94,7 +94,7 @@ public static string FormatValue(object v)
 				return ((Array)v).Cast<object>().Select(FormatValue).JoinWith(", ");
 			}
 
-			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(HashSet<>))
+			if (t.IsGenericType && (t.GetGenericTypeDefinition() == typeof(HashSet<>) || t.GetGenericTypeDefinition() == typeof(List<>)))
 			{
```

Comment:
while writing this PR I observed this method has a number of single line if's, which I believe violates the coding standard. I also noticed above this method the previous type checks could use pattern matching which would also reduce the line count.

For example this looks much nicer to me.
```csharp
			if (v is Rectangle r)
				return "{0},{1},{2},{3}".F(r.X, r.Y, r.Width, r.Height);
```
Would a separate PR using pattern matching be good?

Reply:
I personally feel that some of the weird things that have been introduced around pattern matching in recent C# versions make a real mess of the language consistency, but this does seem like one of the few cases where pattern matching would improve things.